### ISSUE TITLE
HoughLinesP - java.lang.NullPointerException: Attempt to invoke virtual method 'org.bytedeco.javacpp.BytePointer org.bytedeco.javacpp.BytePointer.capacity(long)' on a null object reference

### ISSUE THREAD

**Issue by nicemanis** [bytedeco/javacv#1146]
Hello,
I'm trying to use HoughLinesP method as follows:
```
Mat lines = new Mat();
HoughLinesP(mat_points, lines, 1, CV_PI / 180, 50);
IntRawIndexer linesIdx = lines.createIndexer();
for (long i = 0; i < linesIdx.rows(); i++) {
    int x0 = linesIdx.get(i, 0, 0);
    int y0 = linesIdx.get(i, 0, 1);
    int x1 = linesIdx.get(i, 0, 2);
    int y1 = linesIdx.get(i, 0, 3);
    line(roi, new Point(x0, y0), new Point(x1, y1), Scalar.GREEN);
}
```
However, it throws the following error on this (`IntRawIndexer linesIdx = lines.createIndexer();`) line:
```
java.lang.NullPointerException: Attempt to invoke virtual method 'org.bytedeco.javacpp.BytePointer org.bytedeco.javacpp.BytePointer.capacity(long)' on a null object reference
        at org.bytedeco.javacpp.helper.opencv_core$AbstractMat.createIndexer(opencv_core.java:1881)
        at org.bytedeco.javacpp.helper.opencv_core$AbstractArray.createIndexer(opencv_core.java:147)
```

Is it a bug in the library or am I using it wrong? I tried checking if lines object is null, but it is not.

**Comment by saudet** [comment]
It just looks like HoughLinesP() is failing. Make sure it succeeds before
doing anything more.

**Comment by nicemanis** [comment]
How can I check if it succeeds? Lines object won't be null since I initialized it.
I managed to recreate it with `cvHoughLines2` following [this](https://github.com/bytedeco/javacv/blob/master/samples/HoughLines.java) example. What are the differences between both of these and which method is preferred?

**Comment by saudet** [comment]
Actually, it looks like those functions don't accept a `Mat` for their outputs. I've fixed that in commit https://github.com/bytedeco/javacpp-presets/commit/e1d516dec210faf1380cff1aaea595d9f85e9df2. Please give it a try with 1.4.5-SNAPSHOT in a few hours when the builds are ready: http://bytedeco.org/builds/

**Comment by nicemanis** [comment]
How can I load the snapshot version with Gradle?
If I use:
`implementation group: 'org.bytedeco.javacpp-presets', name: 'opencv-platform', version: '4.0.1-1.4.5-SNAPSHOT'`
It returns
`ERROR: Failed to resolve: org.bytedeco.javacpp-presets:opencv-platform:4.0.1-1.4.5-SNAPSHOT`

**Comment by saudet** [comment]
The instructions for that are on the page:
http://bytedeco.org/builds/

**Comment by saudet** [comment]
The fixes are now included in JavaCV 1.5. Thanks for reporting and enjoy!
